docs/protocol/off-chain/channel-methods.mdx

```diff
@@ -0,0 +1,681 @@
+  "allocations": [
+    {"destination": "0x742d35Cc...", "token": "0x2791Bca1...", "amount": "100000000"},
+    {"destination": "0x123456Cc...", "token": "0x2791Bca1...", "amount": "0"}
+  ],
+  "sigs": [null, "0xClearnodeSignature123..."]
+}
+```
+
+#### StateAllocation {#stateallocation}
+
+| Field | Type | Description |
+|-------|------|-------------|
+| `destination` | wallet address | Participant's wallet address |
+| `token` | wallet address | Token contract address |
+| `amount` | string | Amount in smallest unit (e.g., `"100000000"` for 100 USDC with 6 decimals) |
+
+:::tip Clearnode Signature First
+<Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">The clearnode</Tooltip> provides its signature BEFORE the user commits funds on-chain. This ensures both parties have committed to the channel before any on-chain transaction occurs.
+:::
+
+### Next Steps After Receiving Response
+
+1. **Verify Channel Data**
+   - Recompute `channelId` = `keccak256(abi.encode(channel))`
+   - Verify computed ID matches response `channel_id`
+   - Check participants[0] is your wallet address
+   - Verify token and amount match your request
+
+2. **Verify the Clearnode's Signature**
+   - Compute `stateHash` = `keccak256(abi.encode(channelId, state.intent, state.version, state.data, state.allocations))`
+   - Recover signer from `broker_signature`
+   - Verify signer is <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">the clearnode's</Tooltip> known wallet address
+
+3. **Sign State with Your Key**
+   - Sign `stateHash` with your <Tooltip content="An entity (identified by a wallet address) that is part of a channel">participant</Tooltip> key
+   - Add your signature at `state.sigs[0]`
+
+4. **Approve Token Spending** (if needed)
+   - Check token allowance: `token.allowance(yourAddress, custodyAddress)`
+   - If insufficient: `token.approve(custodyAddress, amount)`
+
+5. **Submit On-Chain Transaction**
+   - Call `Custody.create(channel, state)` with both signatures
+   - Wait for transaction confirmation
+
+6. **Monitor for Channel Opening**
+   - Listen for `Opened` event (emitted when <Tooltip content="A virtual ledger layer that provides a unified ledger (through Nitro RPC) and coordinates state channels (through Nitrolite), providing chain abstraction for developers and users">the clearnode</Tooltip> joins)
+   - Or poll `get_channels` until status is "active"
+
+7. **Channel Active**
+   - Channel appears in <Tooltip content="An abstraction that aggregates a user's funds across multiple blockchain networks, managed by Clearnode">unified balance</Tooltip>
+   - Can now perform off-chain operations
+
+### Error Cases
```

Comment:
Currently we don't have error codes

Reply:
[8211a936738924e9e67a3e56916ec7c6fee618be](https://github.com/layer-3/docs/pull/69/commits/8211a936738924e9e67a3e56916ec7c6fee618be)